editor/find_in_files.cpp

```diff
@@ -874,6 +868,43 @@ void FindInFilesPanel::_on_replace_all_clicked() {
 	emit_signal(SNAME(SIGNAL_FILES_MODIFIED), modified_files);
 }
 
+void FindInFilesPanel::_on_button_clicked(TreeItem *p_item, int p_column, int p_id, int p_mouse_button_index) {
+	const String file_path = p_item->get_text(0);
+
+	if (_result_items.find(p_item)) {
+		_result_items.erase(p_item);
+	}
+
+	if (_file_items.find(file_path)) {
+		TreeItem *file_result = _file_items.get(file_path);
+		int match_count = file_result->get_child_count();
+
+		for (int i = 0; i < match_count; i++) {
+			TreeItem *child_item = file_result->get_child(i);
+
+			if (_result_items.find(child_item)) {
+				_result_items.erase(child_item);
+			}
+		}
+
+		file_result->clear_children();
+		_file_items.erase(file_path);
+	}
+
+	get_tree()->queue_delete(p_item);
```

Comment:
I'd move this below as well just to ensure that we don't operate on something after we've queued the deletion, just for correctness and safety